ten-simple-rules-dockerfiles.Rmd

```diff
@@ -562,70 +561,64 @@ RUN pip install \
-Consider using a suitable package system with pinned versions even for small scripts if the functions are reusable across datasets or workflows by you or others.
-If you cannot publish a software project in the appropriate language's package structure, you should make sure the source code files you `COPY` into the image are published together with the `Dockerfile` (see&nbsp;\ruleref{rule:publish}).
-If the code and `Dockerfile` are in the same repository, the origin and results of the `COPY` instruction and subsequent installation code will be understandable in the future.
-
-Storing _data files_ outside of the container allows for handling of very large datasets and datasets with data worthy of protection, e.g., proprietary data or private information.
-Do not include such data in an image.
+Storing _data files_ outside of the container allows handling of very large or sensitive datasets, e.g., proprietary data or private information.
+Do not include such data in an image!
 To avoid publishing sensitive data by accident, you can add the data directory to the [`.dockerignore`](https://docs.docker.com/engine/reference/commandline/build/#use-a-dockerignore-file) file, which excludes files and directories from the [build context](https://docs.docker.com/engine/reference/commandline/build/#extended-description), i.e., the set of files considered by `docker build`.
-Ignoring data files also speeds up the build in cases where there are very large files or many small files; for example, large data files or the `.git` folder of the [Git version control system](https://git-scm.com/), which contains many small files across numerous folders, are usually not needed during image build.
-In any case, you should include dummy or test data into the image to ensure that a container is functional without the actual dataset, e.g., for automated tests, instructions in the user manual, or peer review (see also "functional testing logic" in @gruening_recommendations_2019).
-For all these cases, you should provide clear instructions for users in the `README` on how to obtain actual or dummy data.
-When publishing your workspace, e.g., on Zenodo, having data and script contents as regular files outside of the container makes them more accessible to others, for example for reuse or analysis.
+Ignoring data files also speeds up the build in cases where there are very large files or many small files.
+You should include dummy or test data in the image to ensure that a container is functional without the actual dataset, e.g., for automated tests, instructions in the user manual, or peer review (see also "functional testing logic" in @gruening_recommendations_2019).
+For all these cases, you should provide clear instructions for users in the `README` file on how to obtain the actual or dummy data.
```

Comment:
For how to _use_ the action or dummy data, and  how to obtain it if it's kept outside of the image?

Reply:
Edited!